flow360/component/simulation/validation/validation_simulation_params.py

```diff
@@ -264,3 +267,55 @@ def _validate_cht_has_heat_transfer(params):
                     )
 
     return params
+
+
+def _check_complete_boundary_condition_and_unknown_surface(params):
+    ## Step 1: Get all boundaries patches from asset cache
+    asset_boundary_entities = params.private_attribute_asset_cache.boundaries
+    if asset_boundary_entities is None or asset_boundary_entities == []:
+        return params
+    asset_boundaries = {boundary.name for boundary in asset_boundary_entities}
+
+    ## Step 2: Collect all used boundaries from the models
+    if len(params.models) == 1 and isinstance(params.models[0], Fluid):
+        return params
+
+    used_boundaries = set()
+
+    for model in params.models:
+        if not isinstance(model, get_args(SurfaceModelTypes)):
+            continue
+
+        entities = []
+        # pylint: disable=protected-access
+        if hasattr(model, "entities"):
+            entities = model.entities._get_expanded_entities(create_hard_copy=False)
+        elif hasattr(model, "entity_pairs"):  # Periodic BC
+            entities = [
+                pair for surface_pair in model.entity_pairs.items for pair in surface_pair.pair
+            ]
+
+        for entity in entities:
+            if isinstance(entity, GhostSurface):
+                continue
+            used_boundaries.add(entity.name)
```

Comment:
should we also check that boundary is used multiple times?

Reply:
It is done in this validator: https://github.com/flexcompute/Flow360/blob/971359b892f2125fa2ae3d1e32ca416efd242112/flow360/component/simulation/validation/validation_simulation_params.py#L57